tests/unit/plugins/modules/net_tools/test_dnsimple_info.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+from ansible_collections.community.general.plugins.modules.net_tools import dnsimple_info
+from ansible_collections.community.general.tests.unit.plugins.modules.utils import AnsibleFailJson, ModuleTestCase, set_module_args, AnsibleExitJson
+from ansible_collections.community.general.tests.unit.compat.mock import patch
+from httmock import response
+from httmock import with_httmock
+from httmock import all_requests
+import pytest
+
+
+dnsimple = pytest.importorskip('dnsimple_info')
+
+
+@all_requests
+def resp(url, request):
+    headers = {'content-type': 'application/json'}
+    data_content = {"data":
+                    [{"dnsimple_*_info":
+                      [{"account_id": "1234", }], }],
+                    "pagination": {"total_pages": 1}}
+    content = data_content
+    return response(200, content, headers, None, 5, request)
+
+
+class TestDNSimple_Info(ModuleTestCase):
+    """Main class for testing dnsimple module."""
+
+    def setUp(self):
+        """Setup."""
+        super(TestDNSimple_Info, self).setUp()
+        self.module = dnsimple_info
+
+    def tearDown(self):
+        """Teardown."""
+        super(TestDNSimple_Info, self).tearDown()
+
+    def test_with_no_parameters(self):
+        """Failure must occurs when all parameters are missing"""
+        with self.assertRaises(AnsibleFailJson):
+            set_module_args({})
+            self.module.main()
+
+    @with_httmock(resp)
+    def test_only_key_and_account(self):
+        """key and account will pass, returns domains"""
+        with self.assertRaises(AnsibleExitJson):
```

Comment:
```suggestion
        with self.assertRaises(AnsibleExitJson) as exc_info:
```
Then you can do
```.py
        result = exc_info.exception.args[0]
        self.assertFalse(result['changed'])
        assert result['xxx'] == 'yyy'
        ...
```

Reply:
That works!! Thanks so much, I had spent hours trying get this in place. I'll add more tests and update asap.